bundles/org.openhab.binding.mikrotik/src/main/java/org/openhab/binding/mikrotik/internal/handler/MikrotikBaseThingHandler.java

```diff
@@ -0,0 +1,226 @@
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.mikrotik.internal.config.ConfigValidation;
+import org.openhab.binding.mikrotik.internal.model.RouterosDevice;
+import org.openhab.core.thing.*;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.builder.ThingStatusInfoBuilder;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link MikrotikBaseThingHandler} is a base class for all other RouterOS things of map-value nature.
+ * It is responsible for handling commands, which are sent to one of the channels and emit channel updates
+ * whenever required.
+ *
+ * @author Oleg Vivtash - Initial contribution
+ *
+ *
+ * @param <C> config - the config class used by this base thing handler
+ *
+ */
+@NonNullByDefault
+public abstract class MikrotikBaseThingHandler<C extends ConfigValidation> extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(MikrotikBaseThingHandler.class);
+    protected @Nullable C config;
+    private @Nullable ScheduledFuture<?> refreshJob;
+    protected LocalDateTime lastModelsRefresh = LocalDateTime.now();
+    protected Map<String, State> currentState = new HashMap<>();
+
+    // public static boolean supportsThingType(ThingTypeUID thingTypeUID) <- in subclasses
+
+    public MikrotikBaseThingHandler(Thing thing) {
+        super(thing);
+    }
+
+    protected @Nullable MikrotikRouterosBridgeHandler getVerifiedBridgeHandler() {
+        @Nullable
+        Bridge bridgeRef = getBridge();
+        if (bridgeRef != null && bridgeRef.getHandler() != null
+                && (bridgeRef.getHandler() instanceof MikrotikRouterosBridgeHandler)) {
+            return (MikrotikRouterosBridgeHandler) bridgeRef.getHandler();
+        }
+        return null;
+    }
+
+    protected final @Nullable RouterosDevice getRouteros() {
+        @Nullable
+        MikrotikRouterosBridgeHandler bridgeHandler = getVerifiedBridgeHandler();
+        return bridgeHandler == null ? null : bridgeHandler.getRouteros();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Handling command = {} for channel = {}", command, channelUID);
+        if (getThing().getStatus() == ONLINE) {
+            RouterosDevice routeros = getRouteros();
+            if (routeros != null) {
+                if (command == REFRESH) {
+                    throttledRefreshModels();
+                    refreshChannel(channelUID);
+                } else {
+                    try {
+                        executeCommand(channelUID, command);
+                    } catch (Exception e) {
```

Comment:
What type of exception do you expect here? Can you specify the concrete type?

Reply:
I'll put `RuntimeException` for now, but generally this is pre-coded for future (the binding does not execute any command except REFRESH at the moment).